gtsfm/evaluation/metrics.py

```diff
@@ -0,0 +1,357 @@
+"""Class to store metrics computed in different GTSfM modules.
+
+Authors: Akshay Krishnan
+"""
+from __future__ import annotations
+
+import json
+from enum import Enum
+from typing import Any, Dict, List, Union
+
+import numpy as np
+
+import gtsfm.utils.io as io
+
+"""Keys to access data in the dictionary of the metric, or the JSON file. 
+   If metric is a distribution, it is saved to JSON in the below format: 
+   metric_name: {
+       DATA_KEY: {
+           .. raw data if stored ..
+       }
+       SUMMARY_KEY: {
+            .. summary (stats) of distribution ..    
+       }
+   }
+   If the metric is scalar, it is stored simply as {metric_name: value}. 
+"""
+DATA_KEY = "full_data"
+SUMMARY_KEY = "summary"
+
+
+class GtsfmMetric:
+    """Class to store a metric computed in a GTSfM module."""
+
+    class PlotType(Enum):
```

Comment:
could you add a brief inline comment explaining why we have plot code here, whenever thing else is agnostic, and doesn't mention visualization? is the idea that we have to choose apriori which type of plot we prefer for each metric, rather than rendering all possible ones, like only one of box plot or histogram would be shown?

Reply:
this could be a good assumption to add as an inline comment, if true

Reply:
I was not too satisfied with the plot type either, its basically used to decide whether to store histogram or quartiles. Let me know if you think theres a better way to do that.